app/main.py

```diff
@@ -1,214 +1,30 @@
-            }
-        ],
-        "weapon": {
-            "name": "Two-handed Sword",
-            "power": 55,
-        },
-        "potion": None,
-    },
-    "mordred": {
-        "name": "Mordred",
-        "power": 30,
-        "hp": 90,
-        "armour": [
-            {
-                "part": "breastplate",
-                "protection": 15,
-            },
-            {
-                "part": "boots",
-                "protection": 10,
-            }
-        ],
-        "weapon": {
-            "name": "Poisoned Sword",
-            "power": 60,
-        },
-        "potion": {
-            "name": "Berserk",
-            "effect": {
-                "power": +15,
-                "hp": -5,
-                "protection": +10,
-            }
-        }
-    },
-    "red_knight": {
-        "name": "Red Knight",
-        "power": 40,
-        "hp": 70,
-        "armour": [
-            {
-                "part": "breastplate",
-                "protection": 25,
-            }
-        ],
-        "weapon": {
-            "name": "Sword",
-            "power": 45
-        },
-        "potion": {
-            "name": "Blessing",
-            "effect": {
-                "hp": +10,
-                "power": +5,
-            }
-        }
-    }
-}
+from app.arenas.camelot import Camelot
+from app.knights.knight import Knight
 
 
-def battle(knightsConfig):
-    # BATTLE PREPARATIONS:
+def battle(knights_config: dict) -> dict:
 
-    # lancelot
-    lancelot = knightsConfig["lancelot"]
+    battles_results = {}
 
-    # apply armour
-    lancelot["protection"] = 0
-    for a in lancelot["armour"]:
-        lancelot["protection"] += a["protection"]
+    battle_arena = Camelot()
+    [Knight(name=knight["name"],
+            hp=knight["hp"],
+            power=knight["power"],
+            armour=knight["armour"],
+            weapon=knight["weapon"],
+            potion=knight["potion"])
+        for knight in knights_config.values()]
 
-    # apply weapon
-    lancelot["power"] += lancelot["weapon"]["power"]
+    first_battle_result = battle_arena.knight_battle(
+        Knight.knights_for_battle.get("Lancelot").prep_for_battle(),
+        Knight.knights_for_battle.get("Mordred").prep_for_battle())
 
-    # apply potion if exist
-    if lancelot["potion"] is not None:
-        if "power" in lancelot["potion"]["effect"]:
-            lancelot["power"] += lancelot["potion"]["effect"]["power"]
+    second_battle_result = battle_arena.knight_battle(
+        Knight.knights_for_battle.get("Arthur").prep_for_battle(),
+        Knight.knights_for_battle.get("Red Knight").prep_for_battle())
 
-        if "protection" in lancelot["potion"]["effect"]:
-            lancelot["protection"] += lancelot["potion"]["effect"]["protection"]
+    battles_results.update(first_battle_result)
 
-        if "hp" in lancelot["potion"]["effect"]:
-            lancelot["hp"] += lancelot["potion"]["effect"]["hp"]
+    battles_results.update(second_battle_result)
```

Comment:
u can just use a key here for update a dict. `update` we use to merge dicts, you don't need it here

Reply:
I decided to use update since my battle_results function returns dictionary and I didn't want to manually type keys/values for each knight, but can change it if necessary